File: src/Common/Desnz.Chmm.ApiClients/Services/RoleService.cs
using Desnz.Chmm.ApiClients.Http;
using Desnz.Chmm.Common.Dtos;
using Microsoft.AspNetCore.Http;

namespace Desnz.Chmm.ApiClients.Services
{
    public class RoleService : HttpServiceClient, IRoleService
    {
        /// <summary>
        /// WARNING: All Uris here MUST NOT have a leading /
        /// </summary>
        public RoleService(HttpClient client, IHttpContextAccessor httpContextAccessor) : base(client, httpContextAccessor)
        {
        }

        public async Task<HttpObjectResponse<List<RoleDto>>> GetAdminRoles()
        {
            return await HttpGetAsync<List<RoleDto>>("api/identity/roles/admin");
        }
    }
}

File: src/Web/Desnz.Chmm.Web/ClientApp/src/app/stores/account-management/services/role.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ChmmRole } from '../dtos/chmm-role';

@Injectable()
export class RoleService {
  private readonly _baseUrl: string;

  constructor(private _httpClient: HttpClient) {
    this._baseUrl = '/api/identity/roles';
  }

  getAdminRoles(): Observable<ChmmRole[]> {
    return this._httpClient.get<ChmmRole[]>(`${this._baseUrl}/admin`, {
      responseType: 'json',
    });
  }
}
